ianalyzer_readers/extract.py

```diff
@@ -73,6 +74,36 @@ def _apply(self, *nargs, **kwargs):
         raise NotImplementedError()
 
 
+    def _is_applicable(self, *nargs, **kwargs) -> bool:
+        '''
+        Checks whether the extractor is applicable, based on the condition passed as the
+        `applicable` parameter.
+        
+        If no condition is provided, this is always true. If the condition is an
+        Extractor, this checks whether the result is truthy.
+
+        If the condition is a callable, it will be called with the document metadata as
+        an argument. This option is deprecated; you can use the Metadata extractor to
+        replace it.
+
+        Raises:
+            TypeError: Raised if the applicable parameter is an unsupported type.
+        '''
+        if self.applicable is None:
+            return True
+        if isinstance(self.applicable, Extractor):
+            return bool(self.applicable.apply(*nargs, **kwargs))
```

Comment:
Won't this check here double the time needed for indexing a given field with `applicable`? In case we have an "expensive" extractor, such as XML tree parsing, this might be an issue. Perhaps good to warn about this in the documentation of the `applicable` argument.

Reply:
Or is the only extractor allowed for the `applicable` argument the `Metadata` extractor? In this case, this is not a worry, but then this should be documented.

Reply:
You mean because there is a processing cost in applying the extractor? If so, yes, that extra processing is added.

But I can't imagine an implementation where you can provide an expensive extractor _without_ incurring its processing cost. How would that even work? Do we need to warn developers that if they provide expensive checks, those checks will actually be run?

I'm genuinely not sure how to formulate such a warning that isn't an obvious statement like "keep in mind that the code you provide needs to be executed as well".

Reply:
Hmm, I see your point. I guess the current setup actually makes the potential processing time more explicit, so perhaps that is in itself already a better documentation than having a callable here, which might be a wrapper around a time consuming XML tree parse, too.

Just a brainfart: Could we make the purpose of what is now implemented as a combination of Backup/Combined and `applicable` even more explicit in a `ConditionalExtractor` class?

Reply:
Something like this?

```python
Condition(
    if=XML('foo'),
    then=XML('bar'),
    else=XML('baz'),
)
```

I can see that working. Or did you have something else in mind?

Reply:
Something like that, indeed. Not necessary to overhaul this PR, but might be more readable in the long run.